In [41]:
import numpy as np
import pandas as pd
import pickle
from sklearn.feature_extraction.text import TfidfTransformer, CountVectorizer
from sklearn.feature_extraction.text import TfidfVectorizer
import scipy.sparse

## Import Data

In [2]:
ekg_denoised = pd.read_pickle('/home/sanjaycollege15/PredictingDiagnoses/Data/ekg_denoised.pkl')
ekg_denoised = ekg_denoised.sample(frac=1)

In [3]:
ekg_denoised.head()

ICD9_CODE                                               TEXT
3515          2  sinus rhythm with occasional atrial and ventri...
2908          1  normal sinus rhythm. marked a-v conduction del...
3209          1  sinus bradycardia with multifocal ventricular ...
495           0  atrial fibrillation with controlled ventricula...
1512          0  lead v3 is absent. sinus tachycardia. generali...

### Train/Test Set

In [14]:
train, test = np.split(ekg_denoised.sample(frac=1, random_state=42), 
                                 [int(.8*len(ekg_denoised))])

## TF-IDF

### Compute IDF Values

In [15]:
tfidf_vectorizer=TfidfVectorizer(use_idf=True) 
tfidf_vectors_train=tfidf_vectorizer.fit_transform(train['TEXT']).astype('float16')

In [40]:
tfidf_vectors_train

<5228x1577 sparse matrix of type '<class 'numpy.float16'>'
	with 127213 stored elements in Compressed Sparse Row format>

In 5228 documents, there are only 1569 unique words. Now we need to apply onto the test set.

In [17]:
tfidf_vectors_test=tfidf_vectorizer.transform(test['TEXT']).astype('float16')

In [18]:
tfidf_vectors_test.shape

(1308, 1577)

### Save tf-idf vectors into dataframe

In [20]:
tfidf_labels_train = np.array(train['ICD9_CODE'])
tfidf_labels_test = np.array(test['ICD9_CODE'])

In [43]:
scipy.sparse.save_npz('/home/sanjaycollege15/PredictingDiagnoses/Data/tfidf_embeddings_train', tfidf_vectors_train)
np.save('/home/sanjaycollege15/PredictingDiagnoses/Data/tfidf_labels_train', tfidf_labels_train)
scipy.sparse.save_npz('/home/sanjaycollege15/PredictingDiagnoses/Data/tfidf_embeddings_test', tfidf_vectors_test)
np.save('/home/sanjaycollege15/PredictingDiagnoses/Data/tfidf_labels_test', tfidf_labels_test)
